src/libraries/System.Formats.Nrbf/src/System/Formats/Nrbf/ArraySinglePrimitiveRecord.cs

```diff
@@ -53,8 +53,26 @@ internal static IReadOnlyList<T> DecodePrimitiveTypes(BinaryReader reader, int c
             return (List<T>)(object)DecodeDecimals(reader, count);
         }
 
+        // char[] has a unique representation in NRBF streams. Typical strings are transcoded
+        // to UTF-8 and prefixed with the number of bytes in the UTF-8 representation. char[]
+        // is also serialized as UTF-8, but it is instead prefixed with the number of chars
+        // in the UTF-16 representation, not the number of bytes in the UTF-8 representation.
+        // This number doesn't directly precede the UTF-8 contents in the NRBF stream; it's
+        // instead contained within the ArrayInfo structure (passed to this method as the
+        // 'count' argument).
+        //
+        // The practical consequence of this is that we don't actually know how many UTF-8
+        // bytes we need to consume in order to ensure we've read 'count' chars. We know that
+        // an n-length UTF-16 string turns into somewhere between [n .. 3n] UTF-8 bytes.
```

Comment:
* > "Is it really 3n? High code points can take 4 UTF-8 code units to encode."
* < "Sure, but those same code points have to be rendered in two UTF-16 code units (surrogate pairs), so that's only 2x."
* > "OK, the highest single-code-unit-in-UTF-16 codepoint is U+FFFF, let's see what that encodes to..."
* > "Hm.  `EF BF BF`.  1 => 3.  So, I guess it is 3n.  Disregard."